apis/v1beta1/providerconfig_types.go

```diff
@@ -24,12 +24,22 @@ import (
 
 // A ProviderConfigSpec defines the desired state of a ProviderConfig.
 type ProviderConfigSpec struct {
-	xpv1.ProviderConfigSpec `json:",inline"`
+	// Credentials required to authenticate to this provider.
+	Credentials ProviderCredentials `json:"credentials"`
 
 	// ProjectID is the project name (not numerical ID) of this GCP ProviderConfig.
 	ProjectID string `json:"projectID"`
 }
 
+// ProviderCredentials required to authenticate.
+type ProviderCredentials struct {
+	// Source of the provider credentials.
+	// +kubebuilder:validation:Enum=None;Secret;InjectedIdentity;Environment;Filesystem
```

Comment:
Should we remove `InjectedIdentity`? I don't think we actually support it in GCP at the moment.

Reply:
@negz I was going to, but we already list it as a valid value, so removing it would technically be a breaking change (i.e. a `ProviderConfig` that used to be "valid" now would not be). Do you think we should still remove it?

Reply:
Ah, good point. That said I'm pretty sure this is a safe change, because it will only affect the way in which we reject this (as yet) unsupported source. If anyone has is trying to use `InjectedIdentity` today it's already not working.